src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {useCallback, useState, useMemo} from 'react'
+import {Text, FlatList, StyleProp, ViewStyle, TextStyle} from 'react-native'
+import AccordionItem from './AccordionItem'
+import type {AnimationType} from './ToggleAnimation'
+
+type ViewStyleProp = StyleProp<ViewStyle> | Array<StyleProp<ViewStyle>>
+type TextTitleStyleProp = StyleProp<TextStyle> | Array<StyleProp<TextStyle>>
+
+export type section = {
+  [key: string]: any
+  title: string
+  content: string
+}
+
+export interface CommonAccordionProps {
+  /**
+   * onPress accordion event
+   */
+  onPress?: (keyExtractorItem: string) => void
+  /**
+   * render custom accordion header
+   */
+  renderHeader?: (
+    item: section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * render custom title section
+   */
+  renderSectionTitle?: (item: section, index: number, expanded: boolean) => React.ReactNode
+  /**
+   * render custom content
+   */
+  renderContent?: (
+    item: section,
+    index: number,
+    expanded: boolean,
+    keyExtractorItem: string,
+  ) => React.ReactNode
+  /**
+   * accordion title
+   */
+  title?: string
+  /**
+   * accordion container style
+   */
+  itemContainerStyle?: ViewStyleProp
+  /**
+   * accordion title style
+   */
+  titleStyle?: TextTitleStyleProp
+  /**
+   * accordion header style
+   */
+  headerContainerStyle?: ViewStyleProp
+  /**
+   * accordion body style
+   */
+  contentContainerStyle?: ViewStyleProp
+  /**
+   * animation type when open
+   * default: easeInEaseOut
+   */
+  openAnimation?: AnimationType
+  /**
+   * animation type when close
+   * default: easeInEaseOut
+   */
+  closeAnimation?: AnimationType
+  /**
+   * duration when open
+   * default: 300
+   */
+  openDuration?: number
+  /**
+   * duration when close
+   * default: 300
+   */
+  closeDuration?: number
+}
+
+export interface AccordionProps extends CommonAccordionProps {
+  /**
+   * section data sets
+   */
+  sections: section[]
+  /**
+   * keyExtractor for Flatlist
+   */
+  keyExtractor?: (item: section, index: number) => string
+  /**
+   * enable expand multiple item
+   * default: false
+   */
+  expandMultiple?: false
+  /**
+   * accordion wrapper style
+   */
+  wrapperStyle?: ViewStyleProp
+}
+
+const Accordion = React.forwardRef<FlatList, AccordionProps>(
+  ({sections, expandMultiple = true, keyExtractor, wrapperStyle, ...props}, ref) => {
+    const [array, setArray] = useState<string[]>([])
+
+    const _keyExtractor = useMemo(
+      () => keyExtractor || ((_: section, index: number) => index.toString()),
+      [keyExtractor],
+    )
+
+    const onPress = (key: string) => {
+      if (expandMultiple) {
+        setArray([key])
+      } else {
+        setArray([...array, key])
+      }
+      if (array.includes(key)) {
+        setArray(array?.filter(el => el !== key))
+      }
+    }
```

Comment:
This function 
- should apply **useCallback** to prevent rendering
- logic used **expandMultiple** is opposite from what it means
- why add key & remove key the same time -> should only remove OR add key

I think it should be:
```suggestion
    const onPress = useCallback((key: string) => {
      if (array.includes(key)) {
        setArray(array?.filter(el => el !== key))
      } else if (expandMultiple) {
          setArray([...array, key])
      } else {
          setArray([key])
      }
    },[array])
```

Reply:
updated